includes/Generator/OrderAttribution.php

```diff
@@ -0,0 +1,321 @@
+<?php
+/**
+ * Order Attribution data helper.
+ *
+ * @package SmoothGenerator\Classes
+ */
+
+namespace WC\SmoothGenerator\Generator;
+
+/**
+ * Order Attribution data helper class.
+ */
+class OrderAttribution {
+
+	/**
+	 * Generate order attribution data.
+	 *
+	 * @param \WC_Order $order Order.
+	 * @param array     $assoc_args Arguments passed via the CLI for additional customization.
+	 */
+	public static function add_order_attribution_meta( $order, $assoc_args = array() ) {
+
+		if ( isset( $assoc_args['skip-order-attribution'] ) ) {
+			return;
+		}
+
+		$device_type    = self::get_random_device_type();
+		$source         = 'woo.com';
+		$source_type    = self::get_source_type();
+		$origin         = self::get_origin( $source_type, $source );
+		$order_products = $order->get_items();
+		$product_url    = get_permalink( $order_products[ array_rand( $order_products ) ]->get_id() );
+		$utm_content    = [ '/', 'campaign_a', 'campaign_b' ];
+		$utm_content    = $utm_content[ array_rand( $utm_content ) ];
+
+		$meta = array();
+
+		// If the source type is admin, we only need to set the source type.
+		if ( 'admin' === $source_type ) {
```

Comment:
To be ready for woocommerce/woocommerce#44519
```suggestion
		// For these source types, we only need to set the source type.
		if ( in_array( $source_type, [ 'admin', 'mobile_app', 'unknown' ], true ) ) {
```